File: LeetCode/LeetCode Practice Problems/389- Find the Difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        map<char,int> m;
        for(int i=0;i<s.length();i++) {
            m[s[i]]++;
        }
        
        for(int i=0;i<t.length();i++) {
            m[t[i]]--;
        }
        
        for(auto i: m) {
            if(i.second != 0) {
                return i.first;
            }
        }
        
        return ' ';
    }
};

File: LeetCode/LeetCode Practice Problems/389-Find the Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int cnt[] = new int[26];
        for(char c: s.toCharArray()) {
            cnt[c - 'a']++;
        }
        for(char c: t.toCharArray()) {
            cnt[c - 'a']--;
        }
        char ch = ' ';
        for(int i=0;i<26;i++) {
            if(cnt[i] == -1) {
                ch = (char)(i + 'a');
                break;
            }
        }
        return ch;
    }
}
